tests/src/test/scala/whisk/core/containerpool/test/ContainerProxyTests.scala

```diff
@@ -761,10 +761,13 @@ class ContainerProxyTests
     override def run(parameters: JsObject, environment: JsObject, timeout: FiniteDuration)(
       implicit transid: TransactionId): Future[(Interval, ActivationResponse)] = {
       runCount += 1
-      environment.fields("api_key") shouldBe message.user.authkey.toJson
+      environment.fields("api_key") shouldBe message.user.authkey.toEnvironment.fields("api_key")
       environment.fields("namespace") shouldBe invocationNamespace.name.toJson
       environment.fields("action_name") shouldBe message.action.qualifiedNameWithLeadingSlash.toJson
       environment.fields("activation_id") shouldBe message.activationId.toJson
+      message.user.authkey.toEnvironment.fields.keySet
+        .filter(environment.fields.contains(_))
+        .size shouldBe message.user.authkey.toEnvironment.fields.size
```

Comment:
This test can be narrowed down a lot, like:

```scala
val authEnvironment = environment.fields.filterKeys(message.user.authkey.toEnvironment.fields.contains)
message.user.authkey.toEnvironment shouldBe authEnvironment
```

Will also compare the values and you can drop the explicit api_key check above.

Reply:
Changed as proposed. This is really shorter !